src/middleware/parser.js

```diff
@@ -0,0 +1,60 @@
+'use strict'
+
+const xml2js = require('xml2js')
+const KoaBodyParser = require('@viweei/koa-body-parser')
+
+const logger = require('../logger')
+const xmlBuilder = new xml2js.Builder()
+
+const parseOutgoingBody = (ctx, outputFormat) => {
+  if (outputFormat === 'XML') {
+    try {
+      ctx.body = xmlBuilder.buildObject(ctx.body)
+      ctx.set('Content-Type', 'application/xml')
+      logger.info(`Parsed outgoing body in ${outputFormat} format`)
+    } catch (error) {
+      throw new Error(`Parsing outgoing body failed: ${error.message}`)
+    }
+  }
+}
+
+const parseIncomingBody = async (ctx, inputFormat, next) => {
+  // parse incoming body
+  // KoaBodyParser executed the next() callback to allow the other middleware to continue before coming back here
+  if (['JSON', 'XML'].includes(inputFormat)) {
+    const options = {
+      limit: '1mb',
```

Comment:
this can be removed because the default limit is 1mb

Reply:
done in 77656a0